src/typescript/tsconfig.json

```diff
@@ -6,7 +6,7 @@
         { "path": "../compiler" },
         { "path": "../jsTyping" },
         { "path": "../services" },
-        { "path": "../deprecatedCompat" }

```

Comment:
Is this intentional to drop drepcated compat from typescript.js  ?

Reply:
It's already in the `server` tsconfig / project, since it's needed for both `tsserver` and `tsserverlibrary`; to make this PR I actually just deleted all of the contents of `typescript` and moved `tsserverlibrary` here, but git unfortunately can't show that!

Reply:
They aren't? But, tsserverlibrary's been set up this way since I merged modules and has been fine; what breaks if this isn't done? If it's about the code itself, it's imported in the `server` namespace so gets pulled in there.

Reply:
Removing `built` and doing `npx tsc -b ./src/typescript`, I see:

![image](https://github.com/microsoft/TypeScript/assets/5341706/59c14aa3-96a4-4e09-ac95-40026e6d4e15)

Then `hereby services` gives me a file with:

![image](https://github.com/microsoft/TypeScript/assets/5341706/4a965d59-cd8f-47ee-bb8e-4d6b49631556)

So, it seems like it all works?

Reply:
i thinjk if you change something in deprecatedCompat, it wont build tsserver library since its not direct dependency unless server needs to be built too. (which is what may be is intended) but just pointing it out. ( i think it made much more difference with prepend, with modules i dont know)

Reply:
Build as in typecheck or as in write out files? I guess dts emit may matter for this...

I'm surprised in that I think a lot of monorepo projects aren't just bulk including all transitive dependencies in their tsconfigs as that would get unweildy.

Reply:
#46153 and #30608

Reply:
Aha! Thanks!

I'll go figure out which transitive things we're missing.

Reply:
I pulled this out into https://github.com/microsoft/TypeScript/pull/55339, which I'd like to merge first.

Reply:
Given this isn't a regression (all I did was move `tsserverlibrary` and rename it), this shouldn't block merging of this PR, right? So the above isn't exactly a prereq?